### ISSUE TITLE
Add action [laterpay_user_based_promotion] to offer trial period

### ISSUE THREAD

**Issue by twoldrich** [laterpay/laterpay-wordpress-plugin#688]
We want to add the option to offer each visitor a **trial period** for a **reduced price** or for **free**, that allows him to access all content on this website.

Therefore, we should add an **action [laterpay_user_based_promotion]**. This action should render a **site notice** like this:

![screenshot 2015-03-11 14 34 11](https://cloud.githubusercontent.com/assets/6638286/6597300/e262b484-c7fb-11e4-8d06-ea08992afeb4.png)

Clicking on this site notice (the **entire** site notice should be the **click target**), should open the purchase dialog for a **time pass**.

There should be three parameters for the action:

Parameter | Description
--------- | -----------
price | Only 0.00 or 0.05 - 5.00 EUR should be possible. Revenue model is always PPU.
period | duration of the trial period in seconds
title | Title for the time pass, that is displayed in the purchase dialog

So if the merchant uses the action

> <?php do_action( ‘laterpay_user_based_promotion’, price=“0”, validity_duration=“3600”, item_title=“1 hour free trial" ); ?>

a user would have access for **1 hour** (3,600 seconds) for **free** (0) and the purchase dialog for this promotion time pass would display "**1 hour free trial**"

A user should be able to purchase such a time pass only once

- [ ] Add action [laterpay_user_based_promotion.]
- [ ] Document action in "Appearance" tab

**Comment by drdla** [comment]
Implementation is on hold. We will probably cancel the feature in this form.

**Comment by drdla** [comment]
@twoldrich Please revise spec.

**Comment by twoldrich** [comment]
What we wanted to do with this feature was to offer the possibility to give each user the possibility to access the content on a website **for free (or a reduced price)** for a **certain period of time**.

After some "rethinking" following our customer's feedback, the easiest solution would probably be to use the **existing time passes** and introduce a simple **"onetime promotion" flag** for **time passes**: 

## What's a "Onetime Promotion" Time Pass?

It should be possible to flag **one** of the existing **time passes** as an "onetime promotion".

This time pass
- will be listed **on top of the time passes list** (if this time pass is available for the respective post).
- can only be purchased **once** per user (as long as this time pass is flagged as "onetime promotion") and won't be displayed to users who already purchased it.
- will be **highlighted** with a green "discount" icon.

![onetime promotion example](https://cloud.githubusercontent.com/assets/6638286/7254198/d40f42ce-e841-11e4-887d-68804b93d89b.png)

## Flagging a Time Pass as "Onetime Promotion"

- When editing a time pass, it should be possible to click on a "promotion" indidcator to flag this time pass as promotion time pass. If another time pass has already been flagged as time pass, this indicator should be greyed-out. 
![onetime promotion-editing](https://cloud.githubusercontent.com/assets/6638286/7254160/9a16ddd4-e841-11e4-9f1f-bb80b5519194.png)

- Hovering over this indicator should display a tooltip to explain, what "onetime promotion" means.
![onetime promotion-hovering](https://cloud.githubusercontent.com/assets/6638286/7254169/a728aa70-e841-11e4-98c4-e1744d38c825.png)

- A time pass flagged as onetime promotion should be highlighted with a green discount icon.
![onetime promotion-preview](https://cloud.githubusercontent.com/assets/6638286/7254189/cbd3f28a-e841-11e4-956f-fa3a9280a79a.png)

@drdla What do you think?

**Comment by twoldrich** [comment]
After talking to @drdla, this functionality should be (with much simpler implementation) be part of a dedicated "promotion" extension, please see Ticket #739.
